app/components/publications/publication/proofs/proof-edit-modal.js

```diff
@@ -0,0 +1,91 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import {
+  isNone, isPresent
+} from '@ember/utils';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import {
+  ValidatorSet, Validator
+} from 'frontend-kaleidos/utils/validators';
+
+/**
+ * @argument {PublicationFlow} publicationFlow
+ * @argument {Piece} piece
+ * @argument onSave
+ * @argument onCancel
+ */
+export default class PublicationsPublicationProofsProofEditModalComponent extends Component {
+  @service currentSession;
+
+  @tracked file = this.args.piece.file;
+  @tracked name = this.args.piece.name;
+  @tracked receivedAtDate = this.args.piece.receivedDate;
+  @tracked publicationSubcaseCorrectionFor;
+
+  validators;
+
+  constructor() {
+    super(...arguments);
+
+    this.init();
+  }
+
+  get isCorrected() {
+    return isPresent(this.publicationSubcaseCorrectionFor);
+  }
+
+  get isReceived() {
+    return isNone(this.publicationSubcaseCorrectionFor);
+  }
+
+  get isCancelDisabled() {
+    return this.save.isRunning;
+  }
+
+  get isSaveDisabled() {
+    return !this.validators.areValid;
+  }
+
+  @task
+  *loadData() {
+    this.publicationSubcaseCorrectionFor = yield this.args.piece.publicationSubcaseCorrectionFor;
+  }
+
+  @task
+  *save() {
+    yield this.args.onSave({
+      file: this.file,
+      name: this.name,
+      receivedAtDate: this.receivedAtDate,
+    });
+  }
+
+  @action
+  setReceivedAtDate(selectedDates) {
+    this.validators.receivedAtDate.enableError();
+
+    if (selectedDates.length) {
+      this.receivedAtDate = selectedDates[0];
+    }
+  }
+
+  async init() {
```

Comment:
`init()` is een method die al gedefinieerd is door het Ember framework op een component. Zie ook https://api.emberjs.com/ember/3.27/classes/Component/methods/init?anchor=init. Je kan hier beter een andere naam kiezen of super callen.

Reply:
Ik weet niet echt een betere naam. `setup` wordt ook al door Ember gebruikt. Glimmer-components hebben wel geen init-methode.

Reply:
Je hebt gelijk. Ik was in `ember/component` aan het kijken :facepalm:  Hou dus maar gewoon `init`.